ai2thor/video_controller.py

```diff
@@ -0,0 +1,336 @@
+"""
+A video controller for ai2thor
+Basic example:
+from ai2thor.controller import VideoController
+with VideoController() as vc:
+    vc.transform(vc.moveAhead())
+    vc.exportVideo('thor.mp4')
+Known issues:
+- Multi agent rotations don't work (since TeleportFull breaks when passing in an AgentID)
+"""
+
+from ai2thor.controller import Controller
+import cv2
+import os
+from PIL import Image
+import scipy.stats as st  # for smooth non-linear animations
+from queue import Queue
+import math
+
+
+class VideoController(Controller):
+    def __init__(self,
+                 initial_camera_rotation=dict(x=85, y=225, z=0),
```

Comment:
mutable default arguments is an anti-pattern:
https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

I'm terribly worried in this case, but its good to consider it.  For instance the following is possible:
```python
a = VideoController()
b = VideoController()
b.initial_camera_rotation['x'] = 100
print(a.initial_camera_rotation['x']) # this will equal 100 as well
c = VideoController()
print(c.initial_camera_rotation['x']) # new instances will equal 100 since the modification was done at the class level

```

The link above provides one solution, another way would be to simply create copy the default argument
```python
self.initial_camera_position = initial_camera_position.copy()
self.initial_camera_rotation = initial_camera_rotation.copy()
```